projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -737,7 +737,11 @@ export class IgxTimePickerComponent extends PickerBaseDirective
             .subscribe(() => this.subToIconsClicked(this.clearComponents, () => this.clear()));
 
         if (this._ngControl) {
-            this._statusChanges$ = this._ngControl.statusChanges.subscribe(this.onStatusChanged.bind(this));
+            this._statusChanges$ =
+            this._ngControl.statusChanges.subscribe(this.onStatusChanged.bind(this));
+            if (this._ngControl.control.validator) {
```

Comment:
From what I gather the asterisk state is not correct on initial render and continues to be until a state change (i.e. make it valid if it starts invalid), right?

Okay, so this is the place. The check for `this._ngControl.control.validator` is a bit redundant - the `this.required` getter already kind of does that, so a simple `this._inputGroup.isRequired = this.required;` will do just as well.

Not sure if you've checked out the date picker's code in this spot - a similar fix was applied there as well with #10170, there's an extra call for detect changes and probably with a good reason. Since at this point (after view init) the input group is checked and done and we're changing its props - you should be seeing "Expression has changed after it was checked" error in the console for this one. Kicking off another change detection should clear that.

Also you can grab the test from that PR and repurpose it for the time picker

Reply:
@Lipata Probably someone from you team should test these PRs are while at it - you might want to check the same behaviors for all form components so they pop up again with the same issue :)

Reply:
I actually used the code from the date picker's ts file but saw that it's working without the additional call for detect changes so I didn't use it, but I'll make it the same if you think that would be better. Same for formatting - no specific reason, I just made it the same as it was in the date picker.